### ISSUE TITLE
QA review 19.09: Batch search  -error from API

### ISSUE THREAD

**Issue by ElaineMcA** [opentargets/platform#734]
On QA, unable to perform batch search either via "upload sample list" or by entering a new list in the paste box.

 
![Screen Shot 2019-09-16 at 13 42 10](https://user-images.githubusercontent.com/40168275/64958904-f3900f80-d887-11e9-9c61-a03d84a52430.png)

![Screenshot 2019-09-16 at 12 15 47](https://user-images.githubusercontent.com/40168275/64958506-40271b00-d887-11e9-8052-45fddf29cb93.png)

![Screenshot 2019-09-16 at 11 50 41](https://user-images.githubusercontent.com/40168275/64958508-40271b00-d887-11e9-9c0f-751c7fdff64a.png)

**Comment by LucaFumis** [comment]
It seems to be working OK at the moment: it might have been a temporary glitch.
The drug sections however has some issues, so I'll look into that and keep an eye on it.
Also note that QA front end hasn't been updated yet - the interface is still from 19.06 (although I don't think this was related).

**Comment by deniseOme** [comment]
When using the Sample List in batch search, we expect having one target returned in the "Suggested matches" box. Target 'mt-nd' is suggested as MT-ND6.

![Screen Shot 2019-09-24 at 09 40 20](https://user-images.githubusercontent.com/6472381/65496146-d70b5d00-deaf-11e9-8a9f-84678e71ebfd.png)

However in two qa instances, we have had 4 additional targets with suggested matches:

<img width="615" alt="Screen Shot 2019-09-23 at 23 18 52" src="https://user-images.githubusercontent.com/6472381/65496201-eb4f5a00-deaf-11e9-95e0-187a777fd1a2.png">

The first qa instance were this was observed was from 16th Sep and the second on the 23rd Sep instance.

**Comment by andrewhercules** [comment]
The issue raised by @deniseOme appears to be related to a change in the response from the `besthitsearch` endpoint.

For the 5 targets under the `Suggested matches` column, the value for `exact` is `false` whereas fr the other 54 targets, the value is `true` - see below for a screenshot

![Screenshot 2019-09-25 at 17 09 44](https://user-images.githubusercontent.com/7490258/65619029-4ddf4d80-dfb7-11e9-8775-da22c1b68afd.png)

@LucaFumis, could you please confirm that this is the issue? If so, we can add as something to fix for 19.11.

**Comment by andrewhercules** [comment]
I have reassigned this to the 19.11 release milestone and it may be resolved with the ongoing search work

**Comment by LucaFumis** [comment]
Had a quick look at this: yes @andrewhercules that seems to be the the issues.
Essentially we use that `exact` field to build the "exact matches" column (with other items going in the "suggested matches" col):
```
    if (thisSearch.result.isExact) { ... }
```

A quick front-end hack is to manually check whether original query and result match - the following seems to do the trick:
```
   if (thisSearch.result.isExact || thisSearch.result.approved_symbol === thisSearch.query) { ... }
```

**Comment by andrewhercules** [comment]
@LucaFumis, could we please include the fix you identified above in the 19.11 release?

**Comment by andrewhercules** [comment]
Ticket closed as issue has been resolved with fix implemented in 19.11 release
